core/src/main/scala/com/salesforce/op/stages/impl/insights/RecordInsightsLOCO.scala

```diff
@@ -74,9 +93,28 @@ class RecordInsightsLOCO[T <: Model[T]]
 
   private lazy val featureInfo = OpVectorMetadata(getInputSchema()(in1.name)).getColumnHistory().map(_.toJson(false))
 
+  private lazy val vectorDummy = Array.fill(featureInfo.length)(0.0).toOPVector
+
+  private[insights] lazy val problemType = modelApply(labelDummy, vectorDummy).score.length match {
+    case 0 => ProblemType.Unknown
+    case 1 => ProblemType.Regression
+    case 2 => ProblemType.BinaryClassification
+    case n if (n > 2) => {
```

Comment:
I was thinking about the multiclass issue -  I think it makes sense to return the difference in score for the winning category so maybe instead of having this find the problem type it should return the index to examine (also move it down to the base score function line 117:

val baseResult = modelApply(labelDummy, features)
val baseScore =  baseResult.score
val indexToExamine = baseScore.length match {
    case 0 => throw new RuntimeExeption("model does not produce scores for insights")
    case 1 => 0
    case 2 => 1
    case n if (n > 2) => baseResult.prediction

and then always use this index for pulling out the diff for the heaps